contracts/starknet/Space.cairo

```diff
@@ -759,26 +761,22 @@ func propose{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr :
 ) -> ():
     alloc_locals
 
-    # We cannot have `0` as the `ethereum_block_number` because we rely on checking
-    # if it's different than 0 in `finalize_proposal`.
-    with_attr error_message("Invalid block number"):
-        assert_not_zero(ethereum_block_number)
-    end
-
     # Verify that the caller is the authenticator contract.
     assert_valid_authenticator()
 
     # Verify that the executor address is one of the whitelisted addresses
     assert_valid_executor(executor)
 
-    let (current_timestamp) = get_block_timestamp()
+    # The snapshot for the proposal is the current timestamp at proposal creation
+    # We use a timestamp instead of a block number to define a snapshot so that the system can generalize to multi-chain
+    let (snapshot_timestamp) = get_block_timestamp()
```

Comment:
Are we absolutely 100% sure `get_block_timestamp` will return something valid (e.g not `0`)? I think yes but I'm just wondering if starknet/ware have some sort of disclaimer somewhere?

Reply:
Yeah I wonder whether it is enforced that the sequencer must include it. Could bring it up with them